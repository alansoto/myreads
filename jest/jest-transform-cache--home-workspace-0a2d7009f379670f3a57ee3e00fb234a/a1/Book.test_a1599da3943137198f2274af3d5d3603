/* istanbul ignore next */'use strict';var _jsxFileName = '/home/workspace/src/Book/Book.test.js';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme'); /* istanbul ignore next */var _enzyme2 = _interopRequireDefault(_enzyme);
var /* istanbul ignore next */_enzymeAdapterReact = require('enzyme-adapter-react-16'); /* istanbul ignore next */var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);
var /* istanbul ignore next */_Book = require('./Book'); /* istanbul ignore next */var _Book2 = _interopRequireDefault(_Book);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

/* istanbul ignore next */_enzyme2.default.configure({ adapter: new /* istanbul ignore next */_enzymeAdapterReact2.default() });

describe('Book Component', () => {
  it('renders without crashing', () => {
    const wrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Book2.default, /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 10 }, __self: undefined }));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders backgroundCover coming from props', () => {
    const wrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Book2.default, /* istanbul ignore next */{ backgroundCover: 'http', __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: undefined }));
    expect(wrapper.find('.book-cover').get(0).props.style.backgroundImage).toBe('url(http)');
  });

  it('renders bookTitle coming from props', () => {
    const wrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Book2.default, /* istanbul ignore next */{ bookTitle: 'Hamlet', __source: { fileName: _jsxFileName, lineNumber: 20 }, __self: undefined }));
    expect(wrapper.find('.book-title').text()).toBe('Hamlet');
  });

  it('renders bookAuthors coming from props', () => {
    const wrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Book2.default, /* istanbul ignore next */{ bookAuthors: 'Shakespeare', __source: { fileName: _jsxFileName, lineNumber: 25 }, __self: undefined }));
    expect(wrapper.find('.book-authors').text()).toBe('Shakespeare');
  });

  it('renders currentBookshelf coming from props', () => {
    const wrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Book2.default, /* istanbul ignore next */{ bookshelf: 'test-bookshelf', __source: { fileName: _jsxFileName, lineNumber: 30 }, __self: undefined }));
    expect(wrapper.find('.book-shelf-changer select').get(0).props.value).toBe('test-bookshelf');
  });

});