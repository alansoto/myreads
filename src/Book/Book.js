import React, {Component} from 'react'
import PropTypes from 'prop-types'

class Book extends Component {
  static propTypes = {
      backgroundCover: PropTypes.string,
      bookTitle: PropTypes.string,
      bookAuthors: PropTypes.string,
      bookshelf: PropTypes.string
  }

  render(){
    const {backgroundCover, bookTitle, bookAuthors, bookshelf} = this.props;

    return (
      <li>
        <div className="book">
          <div className="book-top">
            <div className="book-cover" style={{ width: 128, height: 193, backgroundImage: `url(${backgroundCover})` }}></div>
            <div className="book-shelf-changer">
              <select value={bookshelf}>
                <option value="none" disabled>Move to...</option>
                <option value="currentlyReading">Currently Reading</option>
                <option value="wantToRead">Want to Read</option>
                <option value="read">Read</option>
                <option value="none">None</option>
              </select>
            </div>
          </div>
          <div className="book-title">{bookTitle}</div>
          <div className="book-authors">{bookAuthors}</div>
      	</div>
      </li>
    );
  }
}

export default Book;
