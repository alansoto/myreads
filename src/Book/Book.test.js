/*
import React from 'react';
import ReactDOM from 'react-dom';
import Book from './Book';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Book />, div);
});
*/
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Book from './Book';

Enzyme.configure({adapter: new Adapter()});

it('renders without crashing', () => {
  const wrapper = shallow(<Book />);
  expect(wrapper).toMatchSnapshot();
});

it('has an image property',()=>{
  const wrapper = shallow(<Book bookCover='http'/>)
  
  expect(wrapper.prop('bookCover')).toContain('http');
})
