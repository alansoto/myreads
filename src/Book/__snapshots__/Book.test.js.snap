exports[`test renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Book />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="book">
        <div
                className="book-top">
                <div
                        className="book-cover"
                        style={
                                Object {
                                        "backgroundImage": "url(\"http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api\")",
                                        "height": 193,
                                        "width": 128,
                                      }
                        } />
                <div
                        className="book-shelf-changer">
                        <select>
                                <option
                                        disabled={true}
                                        value="none">
                                        Move to...
                                </option>
                                <option
                                        value="currentlyReading">
                                        Currently Reading
                                </option>
                                <option
                                        value="wantToRead">
                                        Want to Read
                                </option>
                                <option
                                        value="read">
                                        Read
                                </option>
                                <option
                                        value="none">
                                        None
                                </option>
                        </select>
                </div>
        </div>
        <div
                className="book-title">
                Alans Book
        </div>
        <div
                className="book-authors">
                Harper Lee
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="book-top">
            <div
                        className="book-cover"
                        style={
                                    Object {
                                                "backgroundImage": "url(\"http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api\")",
                                                "height": 193,
                                                "width": 128,
                                              }
                        } />
            <div
                        className="book-shelf-changer">
                        <select>
                                    <option
                                                disabled={true}
                                                value="none">
                                                Move to...
                                    </option>
                                    <option
                                                value="currentlyReading">
                                                Currently Reading
                                    </option>
                                    <option
                                                value="wantToRead">
                                                Want to Read
                                    </option>
                                    <option
                                                value="read">
                                                Read
                                    </option>
                                    <option
                                                value="none">
                                                None
                                    </option>
                        </select>
            </div>
</div>,
          <div
            className="book-title">
            Alans Book
</div>,
          <div
            className="book-authors">
            Harper Lee
</div>,
        ],
        "className": "book",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="book-cover"
                style={
                                Object {
                                                "backgroundImage": "url(\"http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api\")",
                                                "height": 193,
                                                "width": 128,
                                              }
                } />,
              <div
                className="book-shelf-changer">
                <select>
                                <option
                                                disabled={true}
                                                value="none">
                                                Move to...
                                </option>
                                <option
                                                value="currentlyReading">
                                                Currently Reading
                                </option>
                                <option
                                                value="wantToRead">
                                                Want to Read
                                </option>
                                <option
                                                value="read">
                                                Read
                                </option>
                                <option
                                                value="none">
                                                None
                                </option>
                </select>
</div>,
            ],
            "className": "book-top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "className": "book-cover",
                "style": Object {
                  "backgroundImage": "url(\"http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api\")",
                  "height": 193,
                  "width": 128,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <select>
                  <option
                                    disabled={true}
                                    value="none">
                                    Move to...
                  </option>
                  <option
                                    value="currentlyReading">
                                    Currently Reading
                  </option>
                  <option
                                    value="wantToRead">
                                    Want to Read
                  </option>
                  <option
                                    value="read">
                                    Read
                  </option>
                  <option
                                    value="none">
                                    None
                  </option>
</select>,
                "className": "book-shelf-changer",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      disabled={true}
                      value="none">
                      Move to...
</option>,
                    <option
                      value="currentlyReading">
                      Currently Reading
</option>,
                    <option
                      value="wantToRead">
                      Want to Read
</option>,
                    <option
                      value="read">
                      Read
</option>,
                    <option
                      value="none">
                      None
</option>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Move to...",
                      "disabled": true,
                      "value": "none",
                    },
                    "ref": null,
                    "rendered": "Move to...",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Currently Reading",
                      "value": "currentlyReading",
                    },
                    "ref": null,
                    "rendered": "Currently Reading",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Want to Read",
                      "value": "wantToRead",
                    },
                    "ref": null,
                    "rendered": "Want to Read",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Read",
                      "value": "read",
                    },
                    "ref": null,
                    "rendered": "Read",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "None",
                      "value": "none",
                    },
                    "ref": null,
                    "rendered": "None",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Alans Book",
            "className": "book-title",
          },
          "ref": null,
          "rendered": "Alans Book",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Harper Lee",
            "className": "book-authors",
          },
          "ref": null,
          "rendered": "Harper Lee",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="book">
          <div
                    className="book-top">
                    <div
                              className="book-cover"
                              style={
                                        Object {
                                                  "backgroundImage": "url(\"http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api\")",
                                                  "height": 193,
                                                  "width": 128,
                                                }
                              } />
                    <div
                              className="book-shelf-changer">
                              <select>
                                        <option
                                                  disabled={true}
                                                  value="none">
                                                  Move to...
                                        </option>
                                        <option
                                                  value="currentlyReading">
                                                  Currently Reading
                                        </option>
                                        <option
                                                  value="wantToRead">
                                                  Want to Read
                                        </option>
                                        <option
                                                  value="read">
                                                  Read
                                        </option>
                                        <option
                                                  value="none">
                                                  None
                                        </option>
                              </select>
                    </div>
          </div>
          <div
                    className="book-title">
                    Alans Book
          </div>
          <div
                    className="book-authors">
                    Harper Lee
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="book-top">
              <div
                            className="book-cover"
                            style={
                                          Object {
                                                        "backgroundImage": "url(\"http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api\")",
                                                        "height": 193,
                                                        "width": 128,
                                                      }
                            } />
              <div
                            className="book-shelf-changer">
                            <select>
                                          <option
                                                        disabled={true}
                                                        value="none">
                                                        Move to...
                                          </option>
                                          <option
                                                        value="currentlyReading">
                                                        Currently Reading
                                          </option>
                                          <option
                                                        value="wantToRead">
                                                        Want to Read
                                          </option>
                                          <option
                                                        value="read">
                                                        Read
                                          </option>
                                          <option
                                                        value="none">
                                                        None
                                          </option>
                            </select>
              </div>
</div>,
            <div
              className="book-title">
              Alans Book
</div>,
            <div
              className="book-authors">
              Harper Lee
</div>,
          ],
          "className": "book",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="book-cover"
                  style={
                                    Object {
                                                      "backgroundImage": "url(\"http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api\")",
                                                      "height": 193,
                                                      "width": 128,
                                                    }
                  } />,
                <div
                  className="book-shelf-changer">
                  <select>
                                    <option
                                                      disabled={true}
                                                      value="none">
                                                      Move to...
                                    </option>
                                    <option
                                                      value="currentlyReading">
                                                      Currently Reading
                                    </option>
                                    <option
                                                      value="wantToRead">
                                                      Want to Read
                                    </option>
                                    <option
                                                      value="read">
                                                      Read
                                    </option>
                                    <option
                                                      value="none">
                                                      None
                                    </option>
                  </select>
</div>,
              ],
              "className": "book-top",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "book-cover",
                  "style": Object {
                    "backgroundImage": "url(\"http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api\")",
                    "height": 193,
                    "width": 128,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": <select>
                    <option
                                        disabled={true}
                                        value="none">
                                        Move to...
                    </option>
                    <option
                                        value="currentlyReading">
                                        Currently Reading
                    </option>
                    <option
                                        value="wantToRead">
                                        Want to Read
                    </option>
                    <option
                                        value="read">
                                        Read
                    </option>
                    <option
                                        value="none">
                                        None
                    </option>
</select>,
                  "className": "book-shelf-changer",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option
                        disabled={true}
                        value="none">
                        Move to...
</option>,
                      <option
                        value="currentlyReading">
                        Currently Reading
</option>,
                      <option
                        value="wantToRead">
                        Want to Read
</option>,
                      <option
                        value="read">
                        Read
</option>,
                      <option
                        value="none">
                        None
</option>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Move to...",
                        "disabled": true,
                        "value": "none",
                      },
                      "ref": null,
                      "rendered": "Move to...",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Currently Reading",
                        "value": "currentlyReading",
                      },
                      "ref": null,
                      "rendered": "Currently Reading",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Want to Read",
                        "value": "wantToRead",
                      },
                      "ref": null,
                      "rendered": "Want to Read",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Read",
                        "value": "read",
                      },
                      "ref": null,
                      "rendered": "Read",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "None",
                        "value": "none",
                      },
                      "ref": null,
                      "rendered": "None",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Alans Book",
              "className": "book-title",
            },
            "ref": null,
            "rendered": "Alans Book",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Harper Lee",
              "className": "book-authors",
            },
            "ref": null,
            "rendered": "Harper Lee",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
